util/cobrautil/cobrautil.go

```diff
@@ -0,0 +1,14 @@
+package cobrautil
+
+import "github.com/spf13/cobra"
+
+// HideInheritedFlags hides inherited flags
+func HideInheritedFlags(cmd *cobra.Command, hidden ...string) {
+	for _, h := range hidden {
+		// we could use cmd.SetHelpFunc to override the helper
+		// but, it's not enough because we also want the generated
+		// docs to be updated, so we override the flag instead
+		cmd.Flags().String(h, "", "")
+		_ = cmd.Flags().MarkHidden(h)
```

Comment:
would have prefer to use `SetHelpFunc` like:

```go
cmd.SetHelpFunc(func(ccmd *cobra.Command, args []string) {
	for _, h := range hidden {
		_ = ccmd.Flags().MarkHidden(h)
	}
	cmd.SetHelpFunc(nil)
	cmd.HelpFunc()(ccmd, args)
})
```

but it cannot be handled for docs generation unfortunately.